### ISSUE TITLE
Add driver API reentrancy support to UART shim drivers

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#404]
**_Reported by Kuo-Lang Tseng:_**

Global data and device IO registers being set by the driver API can be messed up in the middle of the API call when a concurrent call (from other fiber/task) to the same API is made.

(Imported from Jira ZEP-454)

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

This is on hold until ZEP-503 is root-caused.

**Comment by nashif** [comment]
**_by ethan gao:_**

It's being blocked by issue ZEP-503 now and QA will try to verify it as soon as it's available !

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_ethan gao_ , is the verification of this story still blocked by GH-448?  I see that _Kuo-Lang Tseng_  has lowered the priority of GH-448 to medium with this comment, "Updating the priority to medium as the patch reverting the reentrancy support was merged to unblock the issue for now. Baohong will debug the issue when he is done with other high bugs."  If GH-448 is truly blocking the verification of this story (which is needed for the 1.5 release), then it seems to me the priority of GH-448 needs to be raised back up so it gets fixed.  Please advise.
_Sharron LIU_  _Anagha Jamthe_

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

_Mark Linkmeyer_  GH-448 requires more investigation and debug, before we can pull the reentrancy function back to UART shim driver. I wuild suggest we move both GH-448 and GH-404 to 1.6 so we have more time to investigate the issue.

**Comment by nashif** [comment]
**_by Gajinder Vij:_**

The attempt to make the UART driver re-entrant conflicts with the existing Bluetooth driver.  As UART re-entrancy is not a critical feature for 1.5, the feature has been reverted and a solution will be re-proposed for the next Zephyr release.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Gajinder Vij_ , I think you meant to uncheck v1.5.0 in the Fix Version/s field, so I'm doing it for you now.

**Comment by nashif** [comment]
Related to GH-448
